apps/OboeTester/app/src/main/cpp/util/WaveFileWriter.cpp

```diff
@@ -60,30 +60,34 @@ void WaveFileWriter::writeFormatChunk() {
     writeIntLittle(mFrameRate * mSamplesPerFrame * bytesPerSample);
     // block align
     writeShortLittle((int16_t) (mSamplesPerFrame * bytesPerSample));
-    writeShortLittle((int16_t) bitsPerSample);
+    writeShortLittle((int16_t) mBitsPerSample);
+}
+
+int32_t WaveFileWriter::getDataSizeInBytes() {
+    if (mFrameCount <= 0) return INT32_MAX;
+    int64_t dataSize = ((int64_t)mFrameCount) * mSamplesPerFrame * mBitsPerSample / 8;
+    return (int32_t)std::max(dataSize, (int64_t)INT32_MAX);
```

Comment:
shouldn't this be std::min?

Reply:
Yes! I fixed it and used od to check the chunk size bytes. They are correct now.